core/libcamera_app.cpp

```diff
@@ -205,6 +290,9 @@ void LibcameraApp::ConfigureViewfinder()
 		configuration_->at(lores_stream_num).bufferCount = configuration_->at(0).bufferCount;
 	}
 
+	if (select_mode)
+		options_->viewfinder_mode = selectModeForFramerate(size, options_->framerate.value());
+
```

Comment:
I wonder if the `select_mode` flag could disappear if we did this at the top of the function (where we set `select_mode`)? Only a teeny thing, though. Same might apply elsewhere.

Reply:
Can't do this unfortunately, we need the "default" size to be available, and that only happens later on in the function.